omnigibson/systems/micro_particle_system.py

```diff
@@ -999,9 +874,10 @@ def generate_particles_from_link(
             extent = obj.aabb_extent
         # We sample the range of each extent minus
         sampling_distance = 2 * cls.particle_radius if sampling_distance is None else sampling_distance
-        n_particles_per_axis = ((extent - 2 * cls.particle_radius) / sampling_distance).astype(int) + 1
-        arrs = [np.linspace(lo + cls.particle_radius, hi - cls.particle_radius, n) for lo, hi, n in
-                zip(low, high, n_particles_per_axis)]
+        n_particles_per_axis = (extent / sampling_distance).astype(int)
+        assert np.all(n_particles_per_axis), f"link {link.name} is too small to sample any particle of radius {cls.particle_radius}."
+        arrs = [np.arange(lo + cls.particle_radius, hi - cls.particle_radius + 1e-10, cls.particle_radius * 2)
```

Comment:
why is this 1e-10 needed?

Reply:
Oh this is to safeguard the tiny probability that N particles perfectly fit the extent, then the last particle won't be included because (hi - cls.particle_radius) is not inclusive.

Reply:
Oh I see -- could we add a quick comment here?

Reply:
done